### PR TITLE
EVG-6794 log host termination reason in many places

### PR DESCRIPTION
**Pull request by john-m-liu**
There is already a field which displays extra data in the host event log, so this plumbs comments from a variety of places into that field

### CODE REVIEW

**Comment by john-m-liu** [comment]
I could be misunderstanding, but I'm doing so already, such as https://github.com/evergreen-ci/evergreen/pull/2754/files#diff-87b47650b01185ea529331005e3a59c3R287

File: cloud/ec2.go [hunk 1]
```diff
@@ -795,12 +795,12 @@ func (m *ec2Manager) TerminateInstance(ctx context.Context, h *host.Host, user s
 		}
 		// the spot request wasn't fulfilled, so don't attempt to terminate in ec2
 		if instanceId == "" {
-			return errors.Wrap(h.Terminate(user), "failed to terminate instance in db")
+			return errors.Wrap(h.Terminate(user, ""), "failed to terminate instance in db")
```
Comment (tychoish) [inline cloud/ec2.go]: here?

File: cloud/ec2.go [hunk 2]
```diff
@@ -795,12 +795,12 @@ func (m *ec2Manager) TerminateInstance(ctx context.Context, h *host.Host, user s
...
 		}
 	}
 
 	if !strings.HasPrefix(instanceId, "i-") {
-		return errors.Wrap(h.Terminate(user), "failed to terminate instance in db")
+		return errors.Wrap(h.Terminate(user, ""), "failed to terminate instance in db")
```
Comment (tychoish) [inline cloud/ec2.go]: here

File: cloud/ec2.go (hunk 1 above)
Comment (john-m-liu) [reply cloud/ec2.go]: fixed

File: cloud/ec2.go (hunk 2 above)
Comment (john-m-liu) [reply cloud/ec2.go]: fixed

**Pull request merged**